app/stac_api/models.py

```diff
@@ -143,6 +158,16 @@ def validate_item_properties_datetimes(
             properties_end_datetime,
         )
 
+        if properties_datetime is None:
+            if not isinstance(properties_start_datetime, datetime):
+                validate_datetime_string(properties_start_datetime)
+            if not isinstance(properties_end_datetime, datetime):
+                validate_datetime_string(properties_end_datetime)
+            if properties_end_datetime < properties_start_datetime:
+                message = "Property end_datetime can't refer to a date earlier than property "\
+                "start_datetime"
+                raise ValidationError(_(message))
```

Comment:
This should be inside `validate_item_properties_datetimes_dependencies()` as it is also a dependency check. This check won't work if the datetime is a string, the `<` will failed with strings. Instead of using `validate_datetime_string()` we should use the method `utils.fromisoformat()` that return a datetime object. This function raises a `ValueError` in case of invalid string, therefore the code should be wrapped around with a `try: ...; except ValueError as error; raise ValidationError() from error`